src/core.ts

```diff
@@ -124,6 +126,24 @@ export function instrument(eventState: IInstrumentationEventState) {
   }
 }
 
+export function onWorkerMessage(evt: MessageEvent) {
+  let message = JSON.parse(evt.data) as IWorkerMessage;
+  switch (message.type) {
+    case WorkerMessageType.Upload:
+      let uploadMsg = message as IUploadMessage;
+      let onSuccess = (status: number) => { mapProperties(droppedPayloads, uploadDroppedPayloadsMappingFunction, true); };
+      let onFailure = (status: number) => { onFirstSendDeliveryFailure(status, uploadMsg.rawData, uploadMsg.compressedData); };
+      let compressedKb = Math.ceil(uploadMsg.compressedData.length / 1024.0);
+      let rawKb = Math.ceil(uploadMsg.rawData.length / 1024.0);
+      let envelope = JSON.parse(uploadMsg.rawData).envelope as IEnvelope;
+      upload(uploadMsg.compressedData, onSuccess, onFailure);
+      debug(`** Clarity #${envelope.sequenceNumber}: Uploading ${compressedKb}KB (raw: ${rawKb}KB). **`);
+      break;
+    default:
```

Comment:
>default [](start = 4, length = 7)

I think we should add some instrumentation to alert us if the worker is every giving us payloads that we aren't handling

Reply:
How could this ever happen ?

---
In reply to: [136671497](https://github.com/Microsoft/clarity-js/pull/28#discussion_r136671497) [](ancestors = 136671497)

Reply:
It shouldn't, but that's why we'd want to know if it did imo

---
In reply to: [136672665](https://github.com/Microsoft/clarity-js/pull/28#discussion_r136672665) [](ancestors = 136672665,136671497)

Reply:
There is no other source to invoke this message handler, except from the code written in our own compression worker. I'd argue that placing a check here would be quite redundant

---
In reply to: [136673207](https://github.com/Microsoft/clarity-js/pull/28#discussion_r136673207) [](ancestors = 136673207,136672665,136671497)

Reply:
These sorts of comments are meant to make the codebase more robust against future changes. Even when you can prove that an unexpected state can't be reached in the current codebase - you can't guarantee that future changes won't unintentionally break your assumptions, especially in an open source project

---
In reply to: [136676138](https://github.com/Microsoft/clarity-js/pull/28#discussion_r136676138) [](ancestors = 136676138,136673207,136672665,136671497)